install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml

```diff
@@ -65,12 +69,20 @@ spec:
           - --controlPlaneAuthPolicy
           - MUTUAL_TLS
           - --discoveryAddress
-          - istio-pilot.istio-system:15005
+          {{- if $.Values.global.istioNamespace }}
+          - istio-pilot.{{ $.Values.global.istioNamespace }}:15005
```

Comment:
@linsun I was a bit confused here, why using 15005 for both mTLS and non-mTLS mode?

Reply:
I created a PR at https://github.com/istio/istio/pull/7520 to fix this.